src/util/retry.rs

```diff
@@ -28,6 +28,27 @@ use std::time::Duration;
 ///
 /// let result = retry(operation, &retry_policy);
 ///```
+///
+/// To use a retry policy as a member in a struct, we can specify its concrete type as follows:
+/// ```
+/// # use std::time::Duration;
+/// # use vss_client::error::VssError;
+/// # use vss_client::util::retry::{ExponentialBackoffRetryPolicy, FilteredRetryPolicy, retry, RetryPolicy};
+///
+/// type VssRetryPolicy = FilteredRetryPolicy<ExponentialBackoffRetryPolicy<VssError>, Box<dyn 'static + Send + Sync + Fn(&VssError) -> bool>>;
```

Comment:
I think `Box<dyn 'static + Send + Sync + Fn(&VssError) -> bool>` is not generally needed. For example, the following also works:

```diff
diff --git a/src/util/retry.rs b/src/util/retry.rs
index 4f5633c..1dc63fd 100644
--- a/src/util/retry.rs
+++ b/src/util/retry.rs
@@ -35,7 +35,7 @@ use std::time::Duration;
 /// # use vss_client::error::VssError;
 /// # use vss_client::util::retry::{ExponentialBackoffRetryPolicy, FilteredRetryPolicy, retry, RetryPolicy};
 ///
-/// type VssRetryPolicy = FilteredRetryPolicy<ExponentialBackoffRetryPolicy<VssError>, Box<dyn 'static + Send + Sync + Fn(&VssError) -> bool>>;
+/// type VssRetryPolicy = FilteredRetryPolicy<ExponentialBackoffRetryPolicy<VssError>, &'static dyn Fn(&VssError) -> bool>;
 ///
 /// struct SomeStruct {
 ///    retry_policy: VssRetryPolicy,
@@ -43,11 +43,15 @@ use std::time::Duration;
 ///
 /// impl SomeStruct {
 ///    fn new() -> Self {
-///            let retry_policy = ExponentialBackoffRetryPolicy::new(Duration::from_millis(100))
-///                    .skip_retry_on_error(Box::new(|e: &VssError| { matches!( e, VssError::NoSuchKeyError(..)) }) as _);
-///            Self { retry_policy }
+///            Self {
+///                    retry_policy: ExponentialBackoffRetryPolicy::new(Duration::from_millis(100))
+///                            .skip_retry_on_error(&|e| matches!(e, VssError::NoSuchKeyError(..)))
+///            }
 ///    }
 /// }
+///
+/// let s = SomeStruct::new();
+///
 /// ```
 pub async fn retry<R, F, Fut, T, E>(mut operation: F, retry_policy: &R) -> Result<T, E>
 where
```

The actual type parameterization depends on how the struct is needed to be used, IIUC. So might be worth making clear why/when `Box`, `Send`, and `Sync` are needed or reduce the example down a bit, if possible. Or just explicitly state when `&dyn` and `Box<dyn>` are needed.